### PR TITLE
Explicate what "Rc" and "Arc" stand for.

### PR DESCRIPTION
**Pull request by ucarion**
A person on the weekly "Easy Questions" Reddit thread [was mystified by what `Arc`/`Rc` means](https://www.reddit.com/r/rust/comments/6dyud9/hey_rustaceans_got_an_easy_question_ask_here/did87ds/). Though this is explained in various places, it's not mentioned in the documentation directly.

This PR adds an explanation of the `Rc`/`Arc` acronyms to their respective documentations. There are two things I'm not sure of:

* Does "Rc" mean "Reference Count**er**" or "Reference Count**ed**"? ~~I went with the former.~~ *Edit:* I've changed this to use the latter alternative.
* Should this information be spelled out elsewhere, such as in the docs for the `rc` module?

### CODE REVIEW

File: src/liballoc/rc.rs [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 #![allow(deprecated)]
 
-//! Single-threaded reference-counting pointers.
+//! Single-threaded reference-counting pointers. "Rc" stands for "Reference
```
Comment (sfackler) [inline src/liballoc/rc.rs]: I think of Rc as "reference counted", not "reference counter" personally, and Arc as "atomically reference counted".

**Comment by sfackler** [comment]
@rust-lang/docs

File: src/liballoc/rc.rs (hunk 1 above)
Comment (ucarion) [inline src/liballoc/rc.rs]: After reflection, I was reaching the same conclusion. I'll fix the PR.

**Comment by abrugh** [comment]
Hey There. Thanks @ucarion for creating a change request for me. Just commenting to add context. [Here's](https://www.reddit.com/r/rust/comments/6dyud9/hey_rustaceans_got_an_easy_question_ask_here/difmp6f/?context=3 ) the comment in question from reddit.

File: src/liballoc/rc.rs (hunk 1 above)
Comment (nodakai) [inline src/liballoc/rc.rs]: When I compare "atomically reference-counted" with "atomic reference counter," the tech-jargon-ness of the former seems a bit stronger than that of the latter. Just my personal impression.

**Comment by steveklabnik** [comment]
> Does "Rc" mean "Reference Counter" or "Reference Counted"? I went with the former. Edit: I've changed this to use the latter alternative.

Yes, the latter is correct.

> Should this information be spelled out elsewhere, such as in the docs for the rc module?

That might be a good idea, yes 😄

File: src/liballoc/arc.rs [hunk 2]
```diff
@@ -42,7 +42,8 @@ use heap::deallocate;
 /// necessarily) at _exactly_ `MAX_REFCOUNT + 1` references.
 const MAX_REFCOUNT: usize = (isize::MAX) as usize;
 
-/// A thread-safe reference-counting pointer.
+/// A thread-safe reference-counting pointer. "Arc" stands for "Atomically
```
Comment (steveklabnik) [inline src/liballoc/arc.rs]: Both of these should be single quotes

**Review by steveklabnik — CHANGES_REQUESTED** [review]

**Comment by ucarion** [comment]
>>Should this information be spelled out elsewhere, such as in the docs for the rc module?

>That might be a good idea, yes 😄

@steveklabnik I misspoke -- I had added those docs to the `rc` module all along. With the latest commit, I also added docs to the `Rc` struct as well. (I also switched to single quotes)

**Comment by frewsxcv** [comment]
This looks great, thanks!

@bors r+ rollup

**Pull request merged**